lib/doorkeeper/config.rb

```diff
@@ -397,6 +400,10 @@ def configure_secrets_for(type, using:, fallback:)
     option :application_class,
            default: "Doorkeeper::Application"
 
+    # Allows setting a hash of custom data on the OAuth 2.0 Authorization
+    # Server Metadata discovery response.
+    option :custom_discovery_data, default: {}
```

Comment:
This is intended for applications to use, e.g., Mastodon where we want to set `service_documentation` and `app_registration_url` properties in the metadata.